File: 53.maximum-subarray.js
// @leet start
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
	var temp = nums[0];
	var answer = nums[0];
	for (let i = 1; i < nums.length; i++) {
		const num = nums[i];
		temp = Math.max(num, temp + num);
		answer = Math.max(answer, temp);
	}
	return answer
};
// @leet end

File: 53.maximum-subarray.c
// @leet start
int max(int a, int b) {
  if (a < b) {
    return b;
  }
  return a;
}

int maxSubArray(int *nums, int numsSize) {
  int temp = nums[0];
  int answer = nums[0];
  for (int i = 1; i < numsSize; i++) {
    temp = max(nums[i], nums[i] + temp);
    answer = max(answer, temp);
  }
  return answer;
}

// @leet end
